src/transformers/generation/utils.py

```diff
@@ -4287,6 +4298,14 @@ def _ranking_fast(
     norm_context_hidden = context_hidden / context_hidden.norm(dim=2, keepdim=True)
     norm_next_hidden = next_hidden / next_hidden.norm(dim=2, keepdim=True)
     cosine_matrix = torch.matmul(norm_context_hidden, norm_next_hidden.transpose(1, 2)).squeeze(-1)  # [B*K, S]
+
+    if cosine_matrix_mask is not None:
```

Comment:
(this line might need to be removed, depending on the final form of the suggestion above -- i.e. if we're able to create `cosine_matrix_mask` all the times)